### ISSUE TITLE
False positive `no-unwrapped-jsx-text` rule detection when using a literal in conditional rendering

### ISSUE THREAD

**Issue by nokazn** [sayari-analytics/eslint-plugin-sayari#8]
Hi there,
Thank you for creating and maintaining this great plugin. Your plugin helps a lot with preventing React × Google Translate problems.

I found a nit-pick false positive case that I would like to report.

## Current Behavior

When using a `null` literal in conditional rendering, the plugin incorrectly flags it as needing to be wrapped.

**Example:**

```tsx
const Component = () => {
  const someCondition = true;
  return (
    <div>
      <div />
      {/* TODO: implement later */}
      {someCondition && null}
    </div>
  );
};
```

will be fixled like the below.

```tsx
const Component = () => {
  const someCondition = true;
  return (
    <div>
      <div />
      {/* TODO: implement later */}
      {someCondition && <span>null</span>}
    </div>
  );
};
```

This issue seems to occur when the unwrapped node is a literal value, but not when it is a variable:

```tsx
// This case is not reported as an error (works correctly)
const Component = () => {
  const someCondition = true;
  const someValue = null;
  return (
    <div>
      <div />
      {/* TODO: implement later */}
      {someCondition && someValue}
    </div>
  );
};
```

## Expected Behavior

This case should not be modified:

```tsx
const Component = () => {
  const someCondition = true;
  return (
    <div>
      <div />
      {/* TODO: implement later */}
      {someCondition && null}
    </div>
  );
};
```

Appendix: I'm sorry if this issue is duplicated with #1
